pkg/pillar/cmd/volumemgr/sizemgmt.go

```diff
@@ -96,5 +99,58 @@ func dom0DiskReservedSize(ctxPtr *volumemgrContext, deviceDiskSize uint64) uint6
 			"maxDom0DiskSize (%d)", maxDom0DiskSize)
 		diskReservedForDom0 = maxDom0DiskSize
 	}
+	// handle persist/reserved dataset
+	if vault.ReadPersistType() == types.PersistZFS {
+		used, err := zfs.GetDatasetOption(log, types.PersistReservedDataset, "used")
+		if err != nil {
+			log.Errorf("dom0DiskReservedSize: GetDatasetOption failed: %s", err)
+			return diskReservedForDom0
+		}
+		reserved, err := strconv.ParseUint(used, 10, 64)
+		if err != nil {
+			log.Errorf("dom0DiskReservedSize: failed to parse used: %s", err)
+		}
+		diskReservedForDom0 += reserved
```

Comment:
Since Dom0MinDiskUsagePercent is 20% above you end up reseving 40% of the disk, and at most we want the max for /persist/newlog/ which I think is a few GB. So the max of the ZFS reservation and the maxDom0DiskSize makes more sense.

Reply:
OK let's do that separately. One issue is one can't set Dom0MinDiskUsagePercent to zero so the min is 1% which is a lot witth say 100TB storage. We should add a separtate max knob and default it to a few GB.